### PR TITLE
Change ossec.conf to manager.conf and agent.conf in API and Framework

### LINKED ISSUE #7805: Rename ossec.conf to manager.conf and agent.conf in API and Framework
|Wazuh version|Component|
|---|---|
| 5.0.0 | API/Framework |

Working branch: [dev-separate-conf](https://github.com/wazuh/wazuh/tree/dev-separate-conf)

The purpose of this issue is to rename `ossec.conf` in Wazuh API and Framework according to https://github.com/wazuh/wazuh/issues/7787:

- **Agent configuration**: All settings related to Wazuh Agent will be located at `agent.conf`. Blocks:
  - `<active-response>` (client side)
  - `<agent-upgrade>` (client side)
  - `<client>`
  - `<client_buffer>`
  - `<fluent-forward>`
  - `<gcp-pubsub>`
  - `<labels>` (same as manager)
  - `<localfile>`
  - `<logging>` (same as manager)
  - `<rootcheck>`
  - `<sca>`
  - `<socket>`
  - `<syscheck>`
  - `<wodle>`
    - `<wodle name="agent-key-polling">`
    - `<wodle name="aws-s3">`
    - `<wodle name="azure-logs">`
    - `<wodle name="cis-cat">`
    - `<wodle name="command">`
    - `<wodle name="docker-listener">`
    - `<wodle name="open-scap">`
    - `<wodle name="osquery">`
    - `<wodle name="syscollector">`

- **Manager configuration**: All settings related to Wazuh Manager will be located at `manager.conf`. Blocks:
  - `<active-response>` (manager side)
  - `<agentless>`
  - `<agent-upgrade>` (manager side)
  - `<alerts>`
  - `<auth>`
  - `<cluster>`
  - `<command>`
  - `<database_output>`
  - `<email_alerts>`
  - `<global>`
  - `<integration>`
  - `<labels>` (same as agent)
  - `<logging>` (same as agent)
  - `<remote>`
  - `<reports>`
  - `<rule_test>`
  - `<ruleset>`
  - `<syslog_output>`
  - `<task-manager>`
  - `<vulnerability-detector>`

Also, to avoid confusion, `agent.conf` shared configuration file should be renamed to `shared.conf` and `agent-template.conf` file to `shared-template.conf`.

In addition, `gen_ossec.sh` script file has to be renamed to `gen_wazuh.sh`.

WPK agent upgrades should also be tested via API and CLI. This should be the behavior:

- When upgrading using a WPK repository, a version check should be performed:
  - If the agent version is lower than 5.0 and the version to upgrade is greater or equal than 5.0, upgrade should not continue because it is a breaking change (not supported).
  - If the agent version is lower than 5.0 and the version to upgrade is lower than 5.0, upgrade should continue.
- When upgrading using a custom WPK package:
  - If the agent version is lower than 5.0 and the version to upgrade is greater or equal than 5.0, an error will be reported and a roollback has to be done in the agent (check that this is working).
  - If the agent version is lower than 5.0 and the version to upgrade is lower than 5.0, upgrade should continue.

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by mcarmona99**
|Related issue|
|---|
| #7805 |

Hi team,

This PR closes #7805.

I have updated all API and framework functions using `ossec.conf` and references to it using the ossec word as now we will use `manager.conf` and `agent.conf`.
I have also updated old `agent.conf`. The new shared configuration file is called `shared.conf`.

I have also updated the configuration files used in the API integration tests and unittests environments.

A line has also been updated in the `xml_parser.py` as now the configuration files have the tags `wazuh_config` instead of `ossec_config`.

- Wazuh installation working.
- Cluster working.
- API working (some API calls tested).

### Unittests updated:

A lot of changes needed, now we don't have a default `ossec.conf` in the repo's `etc` folder. 
Now we have no template and the `manager/agent.conf` is generated when installing, using `inst-functions.s`h.
I have saved copies of the default `manager.conf` in our unittests data folders.

```
framework/wazuh/core/tests/
============================================== 652 passed, 9 warnings in 2.43s ==============================================
```

For the SDK unittests and others, I have mocked the `wazuh.core.common.manager_conf` variable. The new value is the path to `data/manager_base.conf`. This file is the new `manager.conf `template we are going to use for the unittests.

```
framework/wazuh/tests/
====================== 756 passed, 11 warnings in 52.35s =======================

framework/wazuh/core/cluster/tests/
======================== 55 passed, 4 warnings in 0.38s ========================

framework/wazuh/core/cluster/dapi/tests/
======================== 23 passed, 9 warnings in 0.56s ========================

api/api/test/
======================= 225 passed, 14 warnings in 0.97s =======================
```

Only unit test without changes:

```
framework/wazuh/rbac/test/
======================= 227 passed, 3 warnings in 145.28s (0:02:25) ======================= 
```

TODO:

- wpk agent upgrades changed in API and CLI
- fix API integration test environment, old wazuh agents? changed to jenkins env?

### CODE REVIEW

**Comment by mcarmona99** [comment]
### PR status

I have updated our API testing environment to differentiate between 5.x agents and old agents.

Basic API integration tests fixes and results:

```
test_cluster_endpoints.tavern.yaml 
	 1 failed, 43 passed, 1 xpassed, 47 warnings

The default manager.conf doesn't have cis-cat, rootcheck, syscheck, etc. 
This can be seen in wazuh/etc/templates/config/README.md. It has changed in comparison with <5.x default configuration.
As we can see here https://github.com/wazuh/wazuh/issues/7787, specific agent configuration blocks will be written only in agent.conf. 
That's why manager.conf doesn't have, for instance, syscheck or rootcheck. I have updated expected conf components and filters using them.

test_cluster_endpoints.tavern.yaml 
	 44 passed, 1 xpassed, 47 warnings
```
	 
	 
```
test_agent_DELETE_endpoints.tavern.yaml 
	 6 passed, 8 warnings
```

```
test_agent_POST_endpoints.tavern.yaml 
	 5 passed, 7 warnings
```
	 

```
test_security_DELETE_endpoints.tavern.yaml 
	 15 passed, 17 warnings
```


```
test_security_GET_endpoints.tavern.yaml 
	 18 passed, 20 warnings
```


```
test_security_POST_endpoints.tavern.yaml 
	 7 passed, 9 warnings
```


```
test_security_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
```
	 
	 
```
test_experimental_endpoints.tavern.yaml 
	 5 failed, 5 passed, 1 xfailed, 13 warnings

Agent 000 doesn't have syscollector modules anymore as the manager_conf doesn't include it. 
I have removed all references to 000 in tests and tests expecting 9 items are expecting 8 now.

test_experimental_endpoints.tavern.yaml 
	 10 passed, 1 xpassed, 13 warnings
```

### To do:

- Agent upgrades via wpk changed in API and CLI
- Review:

```
test_agent_PUT_endpoints.tavern.yaml 
	 3 failed, 7 passed, 12 warnings
1- first reconnect test returning 404
2- bad request when restarting agents in an empty group (fixed here https://github.com/wazuh/wazuh/pull/8123/files)
3- update group2 configuration
```


test_agent_GET_endpoints.tavern.yaml 
```
	 17 failed, 74 passed, 93 warnings
	 
Error in GET /groups,
# merged.mg and shared.conf sum
merged_sum = get_hash(path.join(full_entry, "merged.mg"), hash_algorithm)
conf_sum = get_hash(path.join(full_entry, "shared.conf"), hash_algorithm)
conf_sum failing
```

Regards,
Manuel.

**Comment by mcarmona99** [comment]
### PR STATUS

After a rebase,

### Test **agent_PUT**:

```
test_agent_PUT_endpoints.tavern.yaml 
	 1 failed, 9 passed, 12 warnings
```

1- Update group2 configuration
Error: verify-agent-conf hasn't been updated to shared.conf:
```
root@wazuh-master:/# /var/ossec/bin/verify-agent-conf 

verify-agent-conf: Verifying [etc/shared/default/agent.conf]
verify-agent-conf: OK

verify-agent-conf: Verifying [etc/shared/group1/agent.conf]
verify-agent-conf: File not found

verify-agent-conf: Verifying [etc/shared/group3/agent.conf]
verify-agent-conf: File not found

verify-agent-conf: Verifying [etc/shared/group2/agent.conf]
verify-agent-conf: File not found
```


### Test **agent_GET**:

```
test_agent_GET_endpoints.tavern.yaml 
	 17 failed, 74 passed, 93 warnings
```
Same errors as before.

The `default` group doesn't have the new shared.conf (the rest have it as we are adding it manually in our test env):
```
root@wazuh-master:/# ls /var/ossec/etc/shared/default/*conf
/var/ossec/etc/shared/default/agent.conf
root@wazuh-master:/# ls /var/ossec/etc/shared/group*/*conf
/var/ossec/etc/shared/group1/shared.conf  /var/ossec/etc/shared/group2/shared.conf  /var/ossec/etc/shared/group3/shared.conf
```
Error during the installation proccess.



**Working on `Agent upgrades via wpk changed in API and CLI`**

**Comment by mcarmona99** [comment]
### CLI cheks:

```
root@wazuh-master:/# /var/ossec/bin/agent_upgrade -a 001   
Agents that cannot be upgraded:
	Agent 001 upgrade failed. Status: Error 1822 - Current agent version is greater or equal

root@wazuh-master:/# /var/ossec/bin/agent_upgrade -a 008 
Agents that cannot be upgraded:
	Agent 008 upgrade failed. Status: Error 1836 - Upgrading from a version prior to v5.0.0 is a breaking change and is not supported

root@wazuh-master:/# /var/ossec/bin/agent_upgrade -a 008 -v 4.1.0

Upgrading...

Upgraded agents:
	Agent 008 upgraded: Wazuh v3.13.2 -> 4.1.0

root@wazuh-master:/# /var/ossec/bin/cluster_control -a
ID   NAME           IP           STATUS           VERSION        NODE NAME    
000  wazuh-master   127.0.0.1    active           Wazuh v5.0.0   master-node  
001  wazuh-agent1   172.24.0.6   active           Wazuh v5.0.0   worker2      
002  wazuh-agent2   172.24.0.7   active           Wazuh v5.0.0   worker1      
003  wazuh-agent3   172.24.0.8   active           Wazuh v5.0.0   worker2      
004  wazuh-agent4   172.24.0.9   active           Wazuh v5.0.0   worker1      
005  wazuh-agent5   172.24.0.10  active           Wazuh v3.13.2  worker1      
006  wazuh-agent6   172.24.0.11  active           Wazuh v3.13.2  worker1      
007  wazuh-agent7   172.24.0.12  active           Wazuh v3.13.2  worker1      
008  wazuh-agent8   172.24.0.13  active           Wazuh v4.1.0   worker2      
009  wazuh-agent9   any          disconnected     Wazuh v3.9.2   master-node  
010  wazuh-agent10  any          disconnected     Wazuh v3.9.2   master-node  
011  wazuh-agent11  any          never_connected  unknown        unknown      
012  wazuh-agent12  any          never_connected  unknown        unknown    
  
root@wazuh-master:/# /var/ossec/bin/agent_upgrade -a 008 -v 5.0.0
Agents that cannot be upgraded:
	Agent 008 upgrade failed. Status: Error 1836 - Upgrading from a version prior to v5.0.0 is a breaking change and is not supported

root@wazuh-master:/# /var/ossec/bin/agent_upgrade -a 008         
Agents that cannot be upgraded:
	Agent 008 upgrade failed. Status: Error 1836 - Upgrading from a version prior to v5.0.0 is a breaking change and is not supported

```

### API tests:

Updated API integration test with new test cases when trying to upgrade to 5.0.0.

```
test_agent_PUT_endpoints.tavern.yaml 
	 1 failed, 9 passed, 12 warnings
```

Test failing because of the verify-agent-conf error, mentioned in last comment.

**Review by davidjiglesias — APPROVED** [review]

File: framework/wazuh/core/cluster/cluster.json [hunk 1]
```diff
@@ -79,7 +79,8 @@
 
         "excluded_files": [
             "ar.conf",
-            "ossec.conf"
+            "manager.conf",
+            "agent.conf"
```
[the remark this answers is not in the document]
Comment (mcarmona99) [reply framework/wazuh/core/cluster/cluster.json]: Updated this and a unittest that had a test variable that was not being used

**Comment by mcarmona99** [comment]
```
test_rbac_black_agent_endpoints.tavern.yaml 
	 3 failed, 38 passed, 43 warnings

test_rbac_white_agent_endpoints.tavern.yaml 
	 5 failed, 36 passed, 43 warnings
```

Errors related to groups configurations and GET /groups, mentioned above and repeated in GET and PUT agent tests.

**Comment by mcarmona99** [comment]
### Summary:

- Tests excepted to fail:

```
test_agent_GET_endpoints
test_agent_PUT_endpoints
test_rbac_black_agent_endpoints
test_rbac_white_agent_endpoints
```

- Fixed tests after the jenkins execution:

```
test_decoder_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_rbac_black_cluster_endpoints.tavern.yaml 
	 17 passed, 19 warnings

test_rbac_white_cluster_endpoints.tavern.yaml 
	 17 passed, 19 warnings
	 
test_rbac_black_manager_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_manager_endpoints.tavern.yaml 
	 24 passed, 26 warnings

```


`test_rbac_black_manager_endpoints` failed due to a connection error (https://ci.wazuh.info/job/Test_integration_endpoints/552/artifact/artifacts/Test_integration_endpoints_B552_test_rbac_black_manager_endpoints.txt)

**Comment by mcarmona99** [comment]
### Status

I have made a rebase after the changes from agent.conf to shared.conf were made by core.
I have updated our API integration tests and I have used `verify-agent-conf` in our modules as it is the name this binary will have.

I have also updated the manager RBAC and no RBAC API tests as there was a race condition when restarting the manager.

- Test results:

```
test_manager_endpoints.tavern.yaml 
	 24 passed, 26 warnings

test_rbac_black_agent_endpoints.tavern.yaml 
	 41 passed, 43 warnings

test_rbac_black_manager_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_rbac_white_agent_endpoints.tavern.yaml 
	 41 passed, 43 warnings

test_rbac_white_manager_endpoints.tavern.yaml 
	 15 passed, 17 warnings
```

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**